ChronoVisor/include/Chronicle.h

```diff
@@ -154,16 +172,33 @@ class Chronicle {
             LOGD("removing from archiveMap@%p with %lu entries in Chronicle@%p",
                  &archiveMap_, archiveMap_.size(), this);
             auto nErased = archiveMap_.erase(aid);
-            return (nErased == 1);
+            if (nErased == 1) return CL_SUCCESS;
+            else return CL_ERR_UNKNOWN;
         }
-        return false;
+        return CL_ERR_NOT_EXIST;
+    }
+
+    uint64_t incrementAcquisitionCount() {
+        std::lock_guard<std::mutex> lock(statsMutex_);
+        stats_.count++;
+        return stats_.count;
+    }
+    uint64_t decrementAcquisitionCount() {
+        std::lock_guard<std::mutex> lock(statsMutex_);
+        stats_.count--;
+        return stats_.count;
     }
+    uint64_t getAcquisitionCount() const { return stats_.count; }
 
     size_t getPropertyListSize() { return propertyList_.size(); }
     size_t getMetadataMapSize() { return metadataMap_.size(); }
     size_t getStoryMapSize() { return storyMap_.size(); }
     size_t getArchiveMapSize() { return archiveMap_.size(); }
 
+    std::mutex statsMutex_;
```

Comment:
Do you need these three mutexes? stats, attrs, propertyList, metadataMap, storyMap and archiveMap are private members of a chronicle object. In our APIs, chronicle objects are accessed by visor threads through ChronicleMetadirectory (protected  by mutexes). All the members of a chronicle object can be accessed only through these APIs and the chronicle mutex.